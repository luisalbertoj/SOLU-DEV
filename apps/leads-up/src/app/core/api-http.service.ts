import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
