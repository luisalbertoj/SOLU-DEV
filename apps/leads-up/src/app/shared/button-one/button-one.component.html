<button
  pButton
  [type]="type"
  [label]="textContent"
  (click)="handleClick($event)"
  [class.spinner]="loading"
  [disabled]="disabled || loading"
></button>
