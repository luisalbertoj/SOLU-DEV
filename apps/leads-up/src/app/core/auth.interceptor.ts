import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
