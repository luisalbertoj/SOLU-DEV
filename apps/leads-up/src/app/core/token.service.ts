import { Injectable } from '@angular/core';

@Injectable()
export class TokenProvider {
  load() {
    return new Promise((resolve) => {
      resolve(true);
    });
  }
}
