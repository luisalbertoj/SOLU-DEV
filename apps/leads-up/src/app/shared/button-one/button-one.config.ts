import { Directive, EventEmitter, Input, Output } from '@angular/core';

@Directive()
export class ButtonOneConfig {
  /**
   * Name of button
   */
  @Input() key?: string;

  /**
   * Color of button
   * success
   *
   */
  @Input() color?: 'success' | 'primary';

  /**
   * Text of button
   *
   */
  @Input() textContent!: string;

  /**
   * Type of button
   *
   */
  @Input() type?: 'submit' | 'reset' | 'button' = 'button';

  /**
   * Button is disabled
   *
   */
  @Input() disabled?: boolean;

  /**
   * Button is load
   *
   */
  @Input() loading?: boolean;

  /**
   * Event output of button
   */
  @Output() clickButton = new EventEmitter<unknown>();

  /**
   * Event click of button
   */
  @Input() handleClick = ($event: unknown) => {
    this.clickButton.emit({ button: this.key, color: this.color, $event });
    console.log(`Button ${this.key} is clicked`);
  };
}
