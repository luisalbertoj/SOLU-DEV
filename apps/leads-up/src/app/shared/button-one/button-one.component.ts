import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonModule } from 'primeng/button';
import { ButtonOneConfig } from './button-one.config';

@Component({
  selector: 'solu-dev-button-one',
  standalone: true,
  imports: [CommonModule, ButtonModule],
  templateUrl: './button-one.component.html',
  styleUrls: ['./button-one.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonOneComponent extends ButtonOneConfig {}
