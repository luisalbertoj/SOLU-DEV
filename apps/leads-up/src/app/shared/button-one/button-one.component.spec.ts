import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtonOneComponent } from './button-one.component';

describe('ButtonOneComponent', () => {
  let component: ButtonOneComponent;
  let fixture: ComponentFixture<ButtonOneComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ButtonOneComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(ButtonOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
