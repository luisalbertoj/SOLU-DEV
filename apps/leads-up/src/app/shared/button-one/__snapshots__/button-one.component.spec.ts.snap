// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonOneComponent this should be equal to the snapshot 1`] = `"<button pbutton=\\"\\" pripple=\\"\\" class=\\"p-element p-ripple p-button p-component\\" type=\\"button\\"><span class=\\"p-button-label\\">&nbsp;</span></button>"`;
